### ISSUE TITLE
Enum names

### ISSUE THREAD

**Issue by dcharkes** [dart-lang/native#322]
We don't compile C enums to Dart enums, because in Dart we cannot define the integer value of enum items.

However, with our current compilation scheme we don't have access to the string enum names.

```dart

abstract class Cronet_RESULT {
  static const int Cronet_RESULT_SUCCESS = 0;
  static const int Cronet_RESULT_ILLEGAL_ARGUMENT = -100;
  // etc.
}
```

How about we add:

```dart
abstract class Cronet_RESULT {
  // <what we are already generating>

  // Supports having duplicate integers with the same key.
  static const Map<String, int> _nameToValue = {
    'Cronet_RESULT_SUCCESS': 0,
    'Cronet_RESULT_ILLEGAL_ARGUMENT': -100,
    // etc.
  };

  // Does not support having duplicate integers with the same key, picks the (first? last?) name.
  static final Map<int, String> _valueToName = Map.fromEntries(
      _nameToValue.entries.map((e) => MapEntry(e.value, e.key)));

  static String toEnumName(int enumValue) {
    return _valueToName[enumValue]!;
  }

  static int toEnumValue(String enumName) {
    return _nameToValue[enumName]!;
  }
}
```

Any preference on what the static methods should be named?

And of course they should have proper documentation. 😄 

We could make it optional behind a flag (some libraries like sqlite have an API that you pass in the integer and it gives you a full string sentence explaining what's going on, so you might not want to generate the String enum names for that library). But maybe it should be enabled by default.

cc @mannprerak2 @unsuitable001

Context: https://github.com/google/cronet.dart/pull/2/files

More background: `toString()` on enum values in Dart gives you access to the programmatic name: https://stackoverflow.com/questions/29567236/dart-how-to-get-the-value-of-an-enum/29567669

**Comment by Levi-Lesches** [comment]
Closed by #1198?

**Comment by dcharkes** [comment]
> Closed by #1198?

Yep!

Thanks @Levi-Lesches! 🔥
